[docs] [migrate] Docs for YugabyteDB Voyager (#12820)
yugabyte/yugabyte-db: docs/content/preview/migrate/yb-voyager/reference.md

                                              .-----.-------.    | |Import schema | |    .------------------.
                                                                 | .------.-------. |
                                                                 |        |         |
                                                                 |        v         |
                                                                 | .--------------. |
                                                                 | | Import data  | |
                                                                 | .--------------. |
                                                                 |                  |
                                                                 .--------.---------.
```

## Export directory

Before starting migration, you should create the export directory on a file system that has enough space to keep the entire data dump. Next, you should provide the path of the export directory as a mandatory argument (`--export-dir`) to each invocation of the yb-voyager command.

The export directory has the following sub-directories and files:

> - `reports` directory contains the generated Migration Assessment Report.

Author: Yes, Thank you.

- `schema` directory contains the source database schema translated to PostgreSQL. The schema is partitioned into smaller files by the schema object type such as tables, views, and so on.

> - `data` directory contains TSV (Tab Separated Values) files that are passed to the COPY command on the target database.

Author: @ddhodge , I think TSV files get converted to CSV files using the COPY command. @amit-yb , Please confirm.

For most workloads, it is recommended to use HASH partitioning because it efficiently partitions the data, and spreads it evenly across all nodes.

RANGE partitioning may be advantageous for particular use cases. Consider a use case which is time series specific; here you'll be querying data for specific time buckets. In this case, using RANGE partitioning to split buckets into the specific time buckets will help to improve the speed and efficiency of the query.

Additionally, you can use a combination of HASH and RANGE sharding for your primary key by choosing a HASH value as the partition key, and a RANGE value as the clustering key.

## Unsupported features

Currently, yb-voyager doesn't support the following features:

| Feature | Description/Alternatives  | Issue tracker |
| :-------| :---------- | :----------- |
| BLOB and CLOB | yb-voyager currently ignores all columns of type BLOB/CLOB. <br>  Use another mechanism to load the attributes till this feature is supported.| https://github.com/yugabyte/yb-voyager/issues/43 |

> | Tablespaces |  Currently the YB migration engine cannot handle migration of tables associated with certain TABLESPACES automatically. <br> The workaround is to manually create the required tablespace in Yugabyte and then start the migration.<br> Alternatively if that tablespace is not relevant in the YugabyteDB distributed cluster then the user can remove the tablespace association of the table from the create table definition. | https://github.com/yugabyte/yb-voyager/issues/47 |

Reviewer: ```suggestion
| Tablespaces |  Currently YugabyteDB Voyager can't migrate tables associated with certain TABLESPACES automatically. <br> As a workaround, manually create the required tablespace in YugabyteDB and then start the migration.<br> Alternatively if that tablespace is not relevant in the YugabyteDB distributed cluster, you can remove the tablespace association of the table from the create table definition. | https://github.com/yugabyte/yb-voyager/issues/47 |
```
